### DISCUSSION TITLE
Mean misorientations without using grains

### DISCUSSION THREAD

**Discussion by GitHubUserWithoutAName** [mtex-toolbox/mtex#706] [Ask Anything] [answered]
Dear MTEX community,

let's say that I want to calculate the misorientation angle between a single reference orientation `refOri` and a region of interest (ROI) in an EBSD dataset that is not a grain, but which I manually select using e.g. `inpolygon`. I make the point of the ROI not being a grain because grains feature `grains.meanOrientation`, but my custom ROI is just a list of orientations and not stored as a `grain2d`.

There are two things I can do:
1. I first calculate the mean orientation of all EBSD point inside the region of interest using `meanOriROI = mean(<list of all orientations in my ROI>)` and then proceed to calculate the misorientation (angle) between `refOri` and `meanOriROI`.
2. This time, I first calculate the misorientation (angle) between `refOri` and all individual orientations in my ROI, i.e. `arrayOfMisoriAngles = <calculate misorientation angles between refOri and each orientation in ROI>` and then average over the list of misorienation angles using `mean(arrayOfMisoriAngles)`.

At first, I was astonished to find out that the two angles I obtain for methods 1 and 2 are quite different in some (but not all) cases (I looked at a number of test cases, using fcc symmetry). I had expected them to be more or less identical (save some rounding errors, perhaps). After giving this some thought, though, I arrived at the conclusion that neither approach is wrong, but that I just calculate two different things (because the average orientation is not just an arithmetic mean but depends on crystal symmetry etc.) - but I may be wrong. My questions thus are:

1. Would you agree, that both approaches are feasible? I see no alternative to approach 2 if I am interested in the distribution of misorientations, for instance. MTEX tutorials on the other hand frequently seem to use appraoch 1, though, e.g. when looking at misorientations between grains etc.
2. Am I maybe doing something wrong? Is it. e.g. wrong to use `mean()` to obtain the mean orientation of an array of orientations (e.g. because symmetry is not considered, or something like that)? Is grains.meanOrientation calculated differently thanjust performing `mean(`) on the list of all orientations in the grain?

Thank you all very much in advance for any input!

**Comment by ralfHielscher** [comment, marked as answer]
First of all a simple example with integer numbers illustrates that both approaches may lead to different results:
ori = [1,3]; ori_ref = 2; -> ori_mean = 2;
dist(ori_ref,ori_mean) = 0 but mean(dist(ori_ref,ori)) = 1

The difference is that in approach 2 the mean distance includes the variation of orientations within your region of interest while in approach 1 this variation is ignored.


> Would you agree, that both approaches are feasible? I see no alternative to approach 2 if I am interested in the distribution of misorientations, for instance. MTEX tutorials on the other hand frequently seem to use appraoch 1, though, e.g. when looking at misorientations between grains etc.

I think both approaches makes sense.
 
> Am I maybe doing something wrong? Is it. e.g. wrong to use mean() to obtain the mean orientation of an array of orientations (e.g. because symmetry is not considered, or something like that)? Is grains.meanOrientation calculated differently than just performing mean() on the list of all orientations in the grain?

you are not doing anything and everything behaves as you expected.

Ralf.

**Reply by GitHubUserWithoutAName** [reply]
Dear Ralf,

thank you very much for your helpful answer!

Best regards,
GitHubUserWithoutAName
